Release_Process.md

```diff
@@ -42,7 +43,8 @@ These steps are common between minor and patch releases:
    - Target: the release branch that was just pushed
    - Title: Same as tag name
    - Description: The body of the changelog entry (Added, Changed etc)
-1. Publish the release on Github
+1. Publish the release on GitHub
+1. (Optional) Do a `make clean` in the repository root directory (`bigchaindb/`)
```

Comment:
Perhaps, to prevent troubles, it would be best to remove the "(Optional)".

Also, it is important that the PyPI release matches the tagged version in github. For this reason, an extra line, before, the `make clean` instructing to checkout the tag would be necessary, e.g.:

```
$ git fetch <remote-name>
$ git checkout v0.9.1
```
and then `make clean`, `python setup.py sdist`, ...